sql/src/main/java/io/crate/expression/symbol/Field.java

```diff
@@ -60,15 +60,14 @@
 
     private final AnalyzedRelation relation;
     private final Path path;
-    private final DataType valueType;
+    private final Symbol pointer;
```

Comment:
Not sure what to call this - could also be `child`, `source`, `delegate` or just `symbol`.  Although `source` and `child` might be too overloaded as we also use it for the source relations of a relation.

I had also considered changing `Field` to `ScopedSymbol` to promote the fact that a `Field`s purpose is to put a symbol into a different scope. (Might follow up on this if others like this, but wouldn't do it as part of this PR as this is being backported)